rotkehlchen/chain/evm/decoding/merkl/utils.py

```diff
@@ -0,0 +1,156 @@
+                campaign_id = campaign.get('campaignId')
+                protocol_name = opportunity.get('protocol', {}).get('name')
+                if campaign_id and protocol_name:
+                    campaign_to_protocol[campaign_id] = protocol_name
+        except (KeyError, TypeError) as e:
+            error = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+            log.error(f'Failed to process opportunity data due to {error}')
+            continue
+
+    log.debug(f'Retrieved {len(campaign_to_protocol)} Merkl campaign mappings for chain {chain_id}')  # noqa: E501
+    return campaign_to_protocol
+
+
+def query_merkl_rewards(evm_inquirer: 'EvmNodeInquirer', base_tools: BaseDecoderTools) -> None:
+    """Query Merkl rewards for tracked accounts and cache the token-protocol mappings."""
+    if (campaign_to_protocol := _query_merkl_opportunities_by_chain(evm_inquirer.chain_id)) is None:  # noqa: E501
+        log.error(f'Failed to get Merkl campaign mapping for {evm_inquirer.chain_id}, skipping rewards query')  # noqa: E501
+        return
+
+    for account in base_tools.tracked_accounts.get(evm_inquirer.blockchain):
+        if (rewards := _paginated_request(
+            url_template='https://api.merkl.xyz/v4/users/{account}/rewards?chainId={chain_id}&items={items}&page={page}',
+            items_per_page=1000,
+            context=f'Merkl rewards for account {account} on {evm_inquirer.chain_id}',
+            account=account,
+            chain_id=evm_inquirer.chain_id.serialize(),
+        )) is None:
+            continue
+
+        if len(rewards) == 0:
+            log.debug(f'No Merkl rewards found for account {account} on {evm_inquirer.chain_id}')
+            continue
+
+        tokens_protocols = []
+        try:
+            for reward_data in rewards[0]['rewards']:  # we are specifying the chain id, so there will only be a single top level entry  # noqa: E501
+                try:
+                    protocols = set()
+                    reward_token = deserialize_evm_address(reward_data['token']['address'])
+                    for breakdown in reward_data['breakdowns']:
+                        campaign_id = breakdown.get('campaignId')
+                        if campaign_id and campaign_id in campaign_to_protocol:
+                            protocols.add(campaign_to_protocol[campaign_id])
+                        elif campaign_id:
+                            log.debug(f'Campaign ID {campaign_id} not found in opportunities mapping')  # noqa: E501
+
+                    tokens_protocols.append((reward_token, protocols))
+
+                except (DeserializationError, KeyError, TypeError) as e:
+                    error = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+                    log.error(f'Failed to process Merkl reward data for account {account} due to {error}')  # noqa: E501
+                    continue
+
+        except (KeyError, IndexError, TypeError) as e:
```

Comment:
the keyError is only for rewards[0]['rewards'] right? Why not have a check before entering the loop? Same for the index error. What can raise TyperError?